src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/packagist/PackagistParser.groovy

```diff
@@ -68,6 +72,22 @@ class PackagistParser {
         }
         convertFromJsonToDependency(graph, null, startingPackages, packagistPackages, true)
 
```

Comment:
Instead of going through packagistPackages twice (and converting the names/versions twice also) we could simply accumulate some NameVersion instances and then create ExternalId's from those. Might make the code a bit DRY'er.

Reply:
The way it is makes this kind of hard, for now we will take this approach but there is another branch for the better approach that we can do a PR for after 2.3.0